reflect/src/main/java/dagger/reflect/Scope.java

```diff
@@ -69,7 +69,7 @@ private Scope(ConcurrentHashMap<Key, Binding> bindings, JustInTimeLookup.Factory
     if (jitLookup != null) {
       LinkedBinding<?> jitBinding = putJitBinding(key, linker, jitLookup);
       if (jitBinding == null) {
-        throw new IllegalStateException(); // TODO nice error message with scope chain
+        throw new IllegalStateException("Unable to find binding for key=" + key + " and linker=" + linker); // TODO nice error message with scope chain
```

Comment:
Linker has a toString?

Reply:
it doesn't. this error message is mostly to understand if the linker was null or not and what key we were looking for.